### PR TITLE
fix for set transaction of mysql

### LINKED ISSUE #4683: Refer to DB doc to review the rule definition in each g4 file
Although the rule definitions for `MySQL`, `PostgreSQL`, `SQLServer`, `Oracle` and `SQL92` seem complete, it is also necessary to review those rule definitions to make true there are aligned to the definitions from different RDMS.

Note, the official documents of each RDMS are our guidelines.

### PR DESCRIPTION
**Pull request by jingshanglu**
Ref #4683.

Changes proposed in this pull request:
- fix for set transaction of mysql

### CODE REVIEW

File: sharding-sql-test/src/main/resources/sql/supported/tcl/tcl.xml [hunk 1]
```diff
@@ -38,7 +38,7 @@
     <sql-case id="rollbackToSavepoint" value="ROLLBACK TO savepoint1" />
     <sql-case id="setGlobalTransaction" value="SET GLOBAL TRANSACTION ISOLATION LEVEL REPEATABLE READ" />
     <sql-case id="setTransactionReadOnly" value="SET TRANSACTION READ ONLY" db-types="MySQL,Oracle,PostgreSQL" />
-    <sql-case id="setTransactionWithName" value="SET TRANSACTION NAME 'comment1'" />
+    <sql-case id="setTransactionWithName" value="SET TRANSACTION NAME 'comment1'" db-types="Oracle,PostgreSQL,SQLServer"/>
```
Comment (tristaZero) [inline sharding-sql-test/src/main/resources/sql/supported/tcl/tcl.xml:41]: A blank space ahead of `/>` is needed.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/segment/tcl/dialect/mysql/TransactionCharacteristicSegment.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+package org.apache.shardingsphere.sql.parser.sql.segment.tcl.dialect.mysql;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.shardingsphere.sql.parser.sql.segment.SQLSegment;
+
+@Setter
+@Getter
+public class TransactionCharacteristicSegment implements SQLSegment {
```
Comment (tristaZero) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/segment/tcl/dialect/mysql/TransactionCharacteristicSegment.java:26]: Final?

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/antlr4/imports/mysql/TCLStatement.g4 [hunk 3]
```diff
@@ -50,4 +50,16 @@ rollback
 
 savepoint
     : SAVEPOINT
-    ;
\ No newline at end of file
+    ;
+
+transactionCharacteristic_
```
Comment (tristaZero) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/antlr4/imports/mysql/TCLStatement.g4:55]: If you need to visit `transactionCharacteristic_ `, please rename it to `transactionCharacteristic`.
It is also necessary to check other rule names.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/statement/tcl/dialect/mysql/MysqlSetTransactionStatement.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import lombok.Setter;
+import org.apache.shardingsphere.sql.parser.sql.segment.tcl.dialect.mysql.TransactionCharacteristicSegment;
+import org.apache.shardingsphere.sql.parser.sql.statement.tcl.SetTransactionStatement;
+
+import java.util.Collection;
+import java.util.LinkedList;
+
+@Getter
+@Setter
+public final class MysqlSetTransactionStatement extends SetTransactionStatement {
```
Comment (tristaZero) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/statement/tcl/dialect/mysql/MysqlSetTransactionStatement.java:30]: Do other DBs need similar classes?

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/statement/tcl/SetTransactionStatement.java [hunk 5]
```diff
@@ -20,5 +20,5 @@
 /**
  * Set transaction statement.
  */
-public final class SetTransactionStatement extends TCLStatement {
+public class SetTransactionStatement extends TCLStatement {
```
Comment (tristaZero) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/statement/tcl/SetTransactionStatement.java:23]: Is it necessary to create child classes?

**Comment by jingshanglu** [comment]
Well, the key data is placed in SetTransactionStatement, without creating new subclasses to distinguish database types. @tristaZero

**Comment by tristaZero** [comment]
This PR is clear enough, however, there are two of the concerns,
1. Is there any handling for other DBs?
2. How about unit test for your PR?

**Comment by jingshanglu** [comment]
> This PR is clear enough, however, there are two of the concerns,
> 
> 1. Is there any handling for other DBs?
> 2. How about unit test for your PR?

Just for mysql, next subtask will add unit test. @tristaZero

**Pull request merged**